app.js

```diff
@@ -1435,11 +1435,21 @@ app.component('stb-state', {
     }
   },
 
+  methods: {
+    setStartRoutine(ev) {
+      let new_name = ev.target.value;
+      if (new_name == '') {
+        new_name = null;
+      }
+      this.state.start_routine = new_name;
+    },
+  },
+
   template: `
     <div class="stb-state">
       <i class="fas fa-chevron-down" @click="unroll = !unroll" /> <input v-model="state.name" /> <i class="fas fa-trash-alt" @click="$emit('delete')" />
       <ul v-show="unroll">
-        <li><label><span>Start routine:</span> <input v-model="state.start_routine" /></label></li>
+        <li><label><span>Start routine:</span> <input :value="state.start_routine" @change="setStartRoutine" /></label></li>
```

Comment:
`setStartRoutine` can be inlined, avoiding the need of a method, and without making the code unreadable:
```html
<input :value="state.start_routine" @change="state.start_routine = $event.target.value || null" />
```